Formula/a/act_runner.rb

```diff
@@ -0,0 +1,39 @@
+class ActRunner < Formula
+  desc "Action runner for Gitea based on Gitea's fork of act"
+  homepage "https://docs.gitea.com/usage/actions/act-runner"
+  url "https://gitea.com/gitea/act_runner/archive/v0.2.11.tar.gz"
+  sha256 "8b317700d1f3344d8664be9edb004914723a4aacc8f8b1b3719ca2260a5866b6"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    # Upstream uses a Makefile for builds instead of `go build`.
+    #
+    # The `-s -w` LDFLAGS are already provided in the Makefile, and go's linker
+    # doesn't process the `-L/opt/homebrew/opt/ruby/lib` flag correctly (it
+    # complains about the lack of space between the flag and the path), so
+    # LDFLAGS is set to empty here.
+    #
+    # GOFLAGS is used to pass the `-trimpath` flag in `std_go_args` since the
+    # Makefile hardcodes the `go build` invocation. Unfortunately, according to
+    # the `go` command's documentation on environment variables, they get
+    # overridden by the command line flags, so the `-o` flag can't be passed
+    # this way and the binary will need to be copied manually:
+    # https://pkg.go.dev/cmd/go#hdr-Environment_variables
+    system "make", "build", "LDFLAGS=", "GOFLAGS=-trimpath"
+    bin.install "act_runner"
+  end
+
+  test do
```

Comment:
We recommend to add a version check for binary tests.
```suggestion
  test do
    assert_match version.to_s, shell_output("#{bin}/act_runner --version")
```

Reply:
I'm unsure what's going on here, but in one of the environments (`install --build-from-source --interactive` maybe?) `act_runner --version` was returning `act_runner version dev`, and it's currently returning `act_runner version v` from the installed binary on my system. I'll dig into upstream a bit to see what's going on there.

Reply:
Their version string seems to be determined using some Makefile "magic":

```go
// go build -ldflags "-X gitea.com/gitea/act_runner/internal/pkg/ver.version=1.2.3"
var version = "dev"
```

```Makefile
STORED_VERSION_FILE := VERSION

ifneq ($(DRONE_TAG),)
	VERSION ?= $(subst v,,$(DRONE_TAG))
	RELASE_VERSION ?= $(VERSION)
else
	ifneq ($(DRONE_BRANCH),)
		VERSION ?= $(subst release/v,,$(DRONE_BRANCH))
	else
		VERSION ?= main
	endif

	STORED_VERSION=$(shell cat $(STORED_VERSION_FILE) 2>/dev/null)
	ifneq ($(STORED_VERSION),)
		RELASE_VERSION ?= $(STORED_VERSION)
	else
		RELASE_VERSION ?= $(shell git describe --tags --always | sed 's/-/+/' | sed 's/^v//')
	endif
endif

LDFLAGS ?= -X "gitea.com/gitea/act_runner/internal/pkg/ver.version=v$(RELASE_VERSION)"
```

It appears they're using Drone for CI and faking the `DRONE_TAG` variable in the `make` invocation will yield the appropriate version string. Alternatively, writing out the formula version to a file named `VERSION` also achieves the correct result. Is there one of those that would be more preferable/idiomatic?

Reply:
Excellent! Change implemented locally.